sql/session.go

```diff
@@ -724,11 +732,29 @@ func (c *Context) Warn(code int, msg string, args ...interface{}) {
 	})
 }
 
+func (c *Context) KillConnection(connID uint32) error {
+	if c.services.KillConnection != nil {
+		return c.services.KillConnection(connID)
+	}
+	return nil
+}
+
 func (c *Context) NewErrgroup() (*errgroup.Group, *Context) {
 	eg, egCtx := errgroup.WithContext(c.Context)
 	return eg, c.WithContext(egCtx)
 }
 
+// Services are handles to optional or plugin functionality that can be
+// used by the SQL implementation in certain situations. An integrator can set
+// methods on Services for a given *Context and different parts of go-mysql-server
+// will inspect it in order to fulfill their implementations. Currently, the
+// KillConnection service is available. Set these with |WithServices|; the
+// implementation will access them through the corresponding methods on
+// *Context, such as |KillConnection|.
+type Services struct {
```

Comment:
I'm going with this for now...we will see how well it works going forward and we can always change it.

The other short-term addition here is something for the `LOAD DATA` plan node to call into mysql.Conn.HandleLoadDataLocalQuery, which transits the remote file from the client to the server. 
https://github.com/dolthub/vitess/blob/main/go/mysql/conn.go#L739. (`HandleLoadDataLocalQuery` will probably change to something like `func (*Conn) LoadFileFromClient(file string) (io.Reader, error)`)